### PR TITLE
clarify interval and timeout for http health check

### PR DESCRIPTION
**Pull request by opihana**
Fixes https://github.com/hashicorp/consul/issues/2977

See [source code](https://github.com/hashicorp/consul/blob/v0.8.1/command/agent/check.go#L379-L386).

### CODE REVIEW

File: website/source/docs/agent/checks.html.md [hunk 1]
```diff
@@ -50,14 +50,15 @@ There are several different kinds of checks:
...
   fields can be set through the `header` field which is a map of lists of
-  strings, e.g. `{"x-foo": ["bar", "baz"]}`. By default, HTTP checks will be
-  configured with a request timeout equal to the check interval, with a max of
-  10 seconds. It is possible to configure a custom HTTP check timeout value by
-  specifying the `timeout` field in the check definition. The output of the
-  check is limited to roughly 4KB. Responses larger than this will be truncated.
-  HTTP checks also support TLS. By default, a valid TLS certificate is expected.
-  Certificate verification can be turned off by setting the `tls_skip_verify`
-  field to `true` in the check definition.
+  strings, e.g. `{"x-foo": ["bar", "baz"]}`. By default, HTTP checks are set to
```
Comment (banks) [inline website/source/docs/agent/checks.html.md]: This reads a little confusingly, I think you mean the timeout but it sounds more like the Interval.

```suggestion
  strings, e.g. `{"x-foo": ["bar", "baz"]}`. By default, HTTP checks will be configured to timeout after
```

File: website/source/docs/agent/checks.html.md [hunk 2]
```diff
@@ -50,14 +50,15 @@ There are several different kinds of checks:
...
+  10 seconds. If the check interval is less than 10 seconds, the timeout will
+  instead be set to the check interval. This means that the check should return
+  before the next check begins. It is possible to configure a custom HTTP check
+  timeout value by specifying the `timeout` field in the check definition. The
+  output of the check is limited to roughly 4KB. Responses larger than this will
+  be truncated. HTTP checks also support TLS. By default, a valid TLS certificate
+  is expected. Certificate verification can be turned off by setting the
```
Comment (banks) [inline website/source/docs/agent/checks.html.md]: unrelated: we should specify that "valid TLS" means that the CA is trusted by the _system_ trusted root CAs and not the Consul CA if agent TLS is configured.

```suggestion
  trusted by the _system_ certificate authority roots is expected. Certificate verification 
  can be turned off by setting the
```

**Comment by opihana** [comment]
Thanks @banks, how does it look now?

**Comment by pearkes** [comment]
I feel like we can get this merged, @opihana?

**Comment by valentin-krasontovitsch** [comment]
Sorry to give my 2 cents, but I feel like the comment in the source code does not reflect the logic in the source code.

Let me quote the block in question (which conveniently is also linked above):

```
// For long (>10s) interval checks the http timeout is 10s, otherwise the
// timeout is the interval. This means that a check *should* return
// before the next check begins.
if c.Timeout > 0 && c.Timeout < c.Interval {
	c.httpClient.Timeout = c.Timeout
} else if c.Interval < 10*time.Second {
	c.httpClient.Timeout = c.Interval
}
```

The improved docs describe the source code, so I think it would make sense to correct the comment as well (or remove it - some people might say that that's why we have documentation in separate files, so we don't need [to maintain] comments like this) - I personally think it should just be scratched though, the logic is clear from the code.

**Comment by freddygv** [comment]
We're going to change the behavior so that we use the user provided timeout rather than restricting it based on the interval.

See #6094

**Pull request closed without merging**